### PR TITLE
fix(bc-footer): Fix Footer Issues

### PR DESCRIPTION
**Pull request by gzeni-cit**
## Overview
This PR contains some fixes for some issues reported on `feature/bc-footer`:
- Duplicated submenus created;
- Copyright without content has no width, making social links container overlap it;
- H1-H5 in copyright RTE;
- Copyright links have colors mismatching requirements;
- Footer copyright/social links block in wrong position;
- Discrepancy in desktop copyright font size.

We also fix an issue on header:
- When providing content in the utility (same happens to the footer content), it was not applied to the headers globally (logo is fine).

Additionaly, `/styleguide/header` page should contain header only (the same for `styleguide/footer`), so we are providing a fix for that.

## Changes
- Added new divs to the footer to match wireframes specs. Some styles had to be updated because of that;
- To fix footer not updating globally, we apply `withNode` to the footer component as a whole, then we had to update all existing content filenames.
- HeaderOnly/FooterOnly wrappers had issues in the cxStyleGuideTemplate, since they both were using `Default` token, containing `cxLayout.Default` (that contains both Header/Footer) and then adding `cxLayout.Header`/`cxLayout.Footer` in the wrapper on Theme domain. We are now providing cxLayout tokens for specific header and footer to avoid issues and to apply them alone in their pages.

## Test Instructions
N/A

## Related Issues
N/A

### CODE REVIEW

File: packages/cx-layout/src/components/Layout/tokens/cxLayout.ts [hunk 1]
```diff
@@ -29,8 +31,8 @@ import { WithBordersLabels } from './cxLayoutTest';
 const Base = asLayoutToken({
   Components: {
     Helmet: cxHelmet.Default,
-    Header: cxHeader.Default,
-    Footer: cxFooter.Default,
+    Header: replaceWith(Div),
```
Comment (hvanyo) [inline packages/cx-layout/src/components/Layout/tokens/cxLayout.ts]: I think we should change:  layoutComponents: LayoutComponents. Header & footer to be Div and then remove these two lines.

File: packages/cx-layout/src/components/Layout/tokens/cxLayout.ts [hunk 2]
```diff
@@ -72,13 +74,13 @@ const ConstrainSite = asLayoutToken({
 
 const Header = asLayoutToken({
   Components: {
-    Header: as(cxHeader.Default),
+    Header: cxHeader.Default,
```
Comment (hvanyo) [inline packages/cx-layout/src/components/Layout/tokens/cxLayout.ts]: and then update this be on()

File: packages/cx-layout/src/components/Layout/tokens/cxLayout.ts [hunk 3]
```diff
@@ -72,13 +74,13 @@ const ConstrainSite = asLayoutToken({
...
   },
 });
 
 const Footer = asLayoutToken({
   Components: {
-    Footer: as(cxFooter.Default),
+    Footer: cxFooter.Default,
```
Comment (hvanyo) [inline packages/cx-layout/src/components/Layout/tokens/cxLayout.ts]: and then update this be on()

File: packages/cx-editors/src/components/RichText/tokens/cxRichText.ts [hunk 4]
```diff
@@ -109,26 +111,43 @@ const Default = asCxTokenSpec()({
   },
 });
 
-const Copyright = asCxTokenSpec()({
+const Basic = asCxTokenSpec()({
   ...Default,
+  Core: pick(Default.Core, 'paragraph', 'Bold', 'Underline', 'Link', 'SuperScript'),
+  Components: pick(Default.Components, 'Bold', 'Underline', 'SuperScript'),
```
Comment (hvanyo) [inline packages/cx-editors/src/components/RichText/tokens/cxRichText.ts]: Can you change Default.Theme
Then in default change the the Componets -> Theme -- that was my miss.

File: packages/cx-editors/src/components/RichText/tokens/cxRichText.ts [hunk 5]
```diff
@@ -109,26 +111,43 @@ const Default = asCxTokenSpec()({
...
+});
+
+const Copyright = asCxTokenSpec()({
+  ...Basic,
   Spacing: {
     paragraph: 'mx-9 py-9 md:mx-0 md:mb-4 md:p-0 lg:mt-2 lg:mb-0 lg:py-0',
```
Comment (hvanyo) [inline packages/cx-editors/src/components/RichText/tokens/cxRichText.ts]: The spacing should go on the copyright slot in footer not here.

File: packages/cx-editors/src/components/RichText/tokens/cxRichText.ts [hunk 6]
```diff
@@ -109,26 +111,43 @@ const Default = asCxTokenSpec()({
...
   },
   Theme: {
     paragraph: as(
       cxColor.TextPrimaryFooterCopy,
       cxFontSize.XS,
-      'border-white-400 border-t border-b md:border-0 lg:text-m-xs',
+      cxTextDecoration.Normal,
+      'border-white-400 border-t border-b md:border-0',
```
Comment (hvanyo) [inline packages/cx-editors/src/components/RichText/tokens/cxRichText.ts]: The border should go on copyright slot -- we don't want a border between every paragraph.

File: packages/cx-editors/src/components/RichText/tokens/cxRichText.ts [hunk 7]
```diff
@@ -109,26 +111,43 @@ const Default = asCxTokenSpec()({
...
+    ),
+    Link: as(
+      cxColor.TextPrimaryFooterCopy,
+      cxFontSize.XS,
+      cxTextDecoration.Bold,
+      cxTextDecoration.Underline,
+      'text-cx-primary-interactive',
```
Comment (hvanyo) [inline packages/cx-editors/src/components/RichText/tokens/cxRichText.ts]: cxColor.TextPrimaryInteractive

File: packages/cx-layout/src/components/Layout/tokens/cxLayout.ts (hunk 1 above)
Comment (guilhermegpessoa) [reply packages/cx-layout/src/components/Layout/tokens/cxLayout.ts]: Changed Header & Footer to Div and removed the two replaceWith

File: packages/cx-layout/src/components/Layout/tokens/cxLayout.ts (hunk 2 above)
Comment (guilhermegpessoa) [reply packages/cx-layout/src/components/Layout/tokens/cxLayout.ts]: Updated to on()

File: packages/cx-layout/src/components/Layout/tokens/cxLayout.ts (hunk 3 above)
Comment (guilhermegpessoa) [reply packages/cx-layout/src/components/Layout/tokens/cxLayout.ts]: Updated to on()

File: packages/cx-editors/src/components/RichText/tokens/cxRichText.ts (hunk 4 above)
Comment (guilhermegpessoa) [reply packages/cx-editors/src/components/RichText/tokens/cxRichText.ts]: Changed from Components to Theme

File: packages/cx-editors/src/components/RichText/tokens/cxRichText.ts (hunk 5 above)
Comment (guilhermegpessoa) [reply packages/cx-editors/src/components/RichText/tokens/cxRichText.ts]: Spacing changed to the copyright slot

File: packages/cx-editors/src/components/RichText/tokens/cxRichText.ts (hunk 6 above)
Comment (guilhermegpessoa) [reply packages/cx-editors/src/components/RichText/tokens/cxRichText.ts]: Border changed to the copyright slot

File: packages/cx-editors/src/components/RichText/tokens/cxRichText.ts (hunk 7 above)
Comment (guilhermegpessoa) [reply packages/cx-editors/src/components/RichText/tokens/cxRichText.ts]: Added cxColor.TextPrimaryInteractive

**Review by hvanyo — APPROVED** [review]

**Review by hvanyo — APPROVED** [review]

**Pull request merged**